client/src/ui/molecules/offline-status-bar/index.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from "react";
+
+import { isPayingSubscriber } from "../../../utils";
+import { useOnlineStatus } from "../../../hooks";
+import { useUserData } from "../../../user-context";
+
+import "./index.scss";
+
+export function OfflineStatusBar() {
+  const user = useUserData();
+  const onlineStatus = useOnlineStatus();
+  const statusBarRef = React.useRef<HTMLDivElement>(null);
+
+  React.useEffect(() => {
+    const statusBar = statusBarRef.current;
+    const handleOnline = () => {
+      statusBar?.classList.add("is-online");
+    };
+
+    window.addEventListener("online", handleOnline);
+
+    if (statusBar) {
+      // when the fade-out animation ends, remove the class so
+      // the statusbar is hidden not just faded out.
+      statusBar.addEventListener("animationend", () => {
+        statusBar.classList.remove("is-online");
+      });
+    }
+
+    return () => {
+      window.removeEventListener("online", handleOnline);
+    };
+  }, [onlineStatus]);
+
+  if (!isPayingSubscriber(user)) {
```

Comment:
Wonder if it's worth memoizing:

```suggestion
  const hideBar = React.useMemo(() => !isPayingSubscriber(user), [user]);

  if (hideBar) {
```